From dc7a8b3050a2a43f49515a03ae19713dfced75dc Mon Sep 17 00:00:00 2001
From: David Kalnischkies <david@kalnischkies.de>
Date: Sat, 18 Nov 2023 13:10:05 +0000
Subject: [PATCH] Raise cmake_minimum_required to 3.13 to avoid warnings

CMake Deprecation Warning at CMakeLists.txt:6 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

Picking 3.13 here is a semi-random choice to avoid raising the requirement
too much needlessly while also hopefully avoiding needing to raise it
soon again based on the referenced mail.

While we are at it, lets also fix the other spewed warning:

CMake Warning (dev) at CMakeLists.txt:5 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.

References: https://lists.debian.org/msgid-search/20230617162957.6pklb6632zf4nijc@mail.gaussglocke.de

Upstream-Status: Backport [https://salsa.debian.org/apt-team/apt/-/commit/dc7a8b3050a2a43f49515a03ae19713dfced75dc]
Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62182cddf..dae12e7c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,8 @@
 # Licensed under the same terms as APT; i.e. GPL 2 or later.

 # set minimum version
+cmake_minimum_required(VERSION 3.13)
 project(apt)
-cmake_minimum_required(VERSION 3.4.0)
 # Generic header locations
 include_directories(${PROJECT_BINARY_DIR}/include)

--
GitLab
